### ISSUE TITLE
Getting TimeoutException when calling GetMainWindow

### ISSUE THREAD

**Issue by tomaustin700** [FlaUI/FlaUI#211]
Hi,

When trying to run automation tests on a virtual server I am getting the following exception:

System.TimeoutException: UIA Timeout ---> System.Runtime.InteropServices.COMException: Operation timed out. (Exception from HRESULT: 0x80131505)
Stack Trace:
    at Interop.UIAutomationClient.CUIAutomation8Class.ElementFromHandle(IntPtr hwnd)
   at FlaUI.UIA3.UIA3Automation.<>c__DisplayClass27_0.<FromHandle>b__0()
   at FlaUI.Core.Tools.Com.Call[T](Func`1 nativeAction)
--- End of inner exception stack trace ---
    at FlaUI.Core.Tools.Com.Call[T](Func`1 nativeAction)
   at FlaUI.UIA3.UIA3Automation.FromHandle(IntPtr hwnd)

Is there anyway to change the timeout? I have tried setting ConnectionTimeout and TransactionTimeout in the following way however it does not seem to make a difference

automation.ConnectionTimeout = new TimeSpan(0, 5, 0);
 automation.TransactionTimeout = new TimeSpan(0, 5, 0);

Everything works correctly on a physical machine so it could be because the VM is a little slow.

**Comment by Roemer** [comment]
See if #219 helps.

**Comment by tomaustin700** [comment]
@Roemer I have already tried tweaking those values to no avail, see my original post.

**Comment by Roemer** [comment]
What Windows is installed on the VMs? Older Windows do not support those timeouts.

**Comment by tomaustin700** [comment]
Windows 10 1809

**Comment by petrsapak** [comment]
I'm getting exactly the same exception while testing:

W10 1809  - VM                         - UIA Timeout
W10 1809  - Physical machine   - OK
W7             - VM                         - OK
W7             - Physical machine   - OK

Currently I'm trying to tweak the Connection/Transaction Timeouts.

**Comment by tomaustin700** [comment]
@petrsapak What we ended up doing was building our own ```GetMainWindow``` and doing something like this, it's not elegant but does seem to work
```csharp
while (window == null && count <= 500)
{
    try
    {
         window =  automation.FromHandle(mainWindowHandle).AsWindow();
    }
    catch
    {
        count++;
    }       
 }
```
The ```count <=5000``` is just so it doesn't loop forever.

**Comment by Roemer** [comment]
You can use FlaUIs `Retry` for that. There is an option to ignore exceptions and just retry again in case of an exception.
So it would look something like: 
```csharp
Retry.WhileNull(() => automation.FromHandle(mainWindowHandle).AsWindow(), timeout:TimeSpan.FromMinutes(1), ignoreException: true);
```

**Comment by petrsapak** [comment]
@Roemer & @tomaustin700  Thanks guys, I've tries Roemer's proposal and it seems to work!

**Comment by abhijeet-vidhate-spider** [comment]
I ran into this issue. Roemer's suggestion didn't work as it is for me. I had to change it to

`var retryResult = Retry.WhileNull(() => _app.MainWindowHandle != IntPtr.Zero ? automation.FromHandle(_app.MainWindowHandle).AsWindow() : null, timeout: TimeSpan.FromSeconds(Settings.MaxLaunchDelay), ignoreException: true);`

That way, i could get the main window object. But even after that, very first FindFirstDescendant call fails to give me the control I am looking for. Just as an alternative, I tried following code:

```
Thread.Sleep(TimeSpan.FromSeconds(Settings.MaxLaunchDelay));
_mainWindow = _app.GetMainWindow(automation, null);
```
and FindFirstDescendant works after that.

Am I missing something here?
